subsys/fs/ext2/ext2_diskops.h

```diff
@@ -0,0 +1,145 @@
+ * @param ino Inode number
+ * @param inode Buffer for fetched inode
+ */
+int fetch_inode(struct ext2_data *fs, uint32_t ino, struct ext2_inode *inode);
+
+/**
+ * @brief Fetch block into buffer in the inode structure.
+ *
+ * @param inode Inode structure
+ * @param block Number of inode block to fetch (0 - first block in that inode)
+ */
+int fetch_inode_block(struct ext2_inode *inode, uint32_t block);
+
+/**
+ * @brief Fetch block group into buffer in fs structure.
+ *
+ * If the group was already fetched then this function has no effect.
+ * If the group was not fetched but the block that holds this group is present in
+ * the internal buffer, then the block isn't fetched again.
+ *
+ * @param fs File system data
+ * @param group Block group number
+ */
+int fetch_block_group(struct ext2_data *fs, uint32_t group);
+
+/*
+ * @brief Fetch one block of inode table into internal buffer
+ *
+ * If the block of inode table was already fetched then this function has no
+ * effect.
+ *
+ * @param bg Block group structure
+ * @param block Number of inode table block to fetch (relative to start of the inode table)
+ */
+int fetch_bg_itable(struct ext2_bgroup *bg, uint32_t block);
+
+/**
+ * @brief Fetch inode bitmap into internal buffer
+ *
+ * If the bitmap was already fetched then this function has no effect.
+ *
+ * @param bg Block group structure
+ */
+int fetch_bg_ibitmap(struct ext2_bgroup *bg);
+
+/**
+ * @brief Fetch block bitmap into internal buffer
+ *
+ * If the bitmap was already fetched then this function has no effect.
+ *
+ * @param bg Block group structure
+ */
+int fetch_bg_bbitmap(struct ext2_bgroup *bg);
+
+/**
+ * @brief Clear the entry in the inode table of given inode
+ *
+ * This function triggers fetching of block group and inode table (where the
+ * inode is described).
+ *
+ * @param fs File system data
+ * @param inode Inode number of inode to clear
+ */
+int clear_inode(struct ext2_data *fs, uint32_t ino);
+
+/**
+ * @brief Commit changes made to inode structure
+ *
+ * The changes are committed only to the cached block. They are truly written to
+ * storage when sync is performed.
+ *
+ * @param inode Inode structure
+ */
+int commit_inode(struct ext2_inode *inode);
+
+/**
+ * @brief Commit changes made to inode block
+ *
+ * The changes are committed only to the cached block. They are truly written to
+ * storage when sync is performed.
+ *
+ * @param inode Inode structure
+ */
+int commit_inode_block(struct ext2_inode *inode);
+
+
+/* Operations that reserve or free the block or inode in the file system. They
+ * mark an inode or block as used in the bitmap and change free inode/block
+ * count in superblock and block group.
+ */
+
+/**
+ * @brief Reserve a block for future use.
+ *
+ * Search for a free block. If block is found, proper fields in superblock and
+ * block group are updated and block is marked as used in block bitmap.
+ *
+ * @param fs File system data
+ */
+int64_t alloc_block(struct ext2_data *fs);
```

Comment:
Not sure if this would be a problem, but there is no prefix for any of these functions something like "alloc_block" seems like it is going to conflict with other code/libraries and sounds like a very common name

Reply:
I suppose that it will be good to have all the functions in ext2 implementation prefixed with `ext2_`? Do you agree with that?

Reply:
Yes, that'd be great